### ISSUE TITLE
Support committing a single dataset when multiple have changes

### ISSUE THREAD

**Issue by UberMouse** [koordinates/sno#69]

**Comment by craigds** [comment]
tagging this for milestone 0.4, but we need to flesh out exactly what we want it to look like shortly.

As a straw man for discussion, how about 

```
#  commits all of dataset1, and one feature in dataset2
sno commit dataset1 dataset2:fid=1
```


Since `commit` doesn't yet take any positional args in sno or in git, I think this could be a sensible approach.

It would naturally lead into this

```
# commit pieces interactively, like `git add -p`
sno commit -i
sno commit --interactive
```

**Comment by olsen232** [comment]
Started work on this today - committing a partial working copy, works, but then the working-copy has its "tree" field set to the HEAD then it shows as having no diffs.

Not sure how much work this is to fix: maybe it should be in 0.5 as part of the dataset v2.

**Comment by craigds** [comment]
IMO that's probably not particularly hard to fix; we just have to delete particular things from the tracking table instead of truncating the whole table.

The tracking table contains both table_name and pk:
```
sqlite> .schema %track%
CREATE TABLE IF NOT EXISTS ".sno-track" (
                    table_name TEXT NOT NULL,
                    pk TEXT NULL,
                    CONSTRAINT ".sno-track_pk" PRIMARY KEY (table_name, pk)
                );
sqlite>
```

so it seems reasonably straightforward to delete from selectively. The main difficulty of implementing this might just be in refactoring the various layers of python callbacks so the filter information can be passed through to the bit that actually does the delete
